homeassistant/components/zha/binary_sensor.py

```diff
@@ -293,3 +293,11 @@ class AqaraLinkageAlarmState(BinarySensor, id_suffix="linkage_alarm_state"):
     SENSOR_ATTR = "linkage_alarm_state"
     _attr_name: str = "Linkage alarm state"
     _attr_device_class: BinarySensorDeviceClass = BinarySensorDeviceClass.SMOKE
+
+
+@MULTI_MATCH(quirk_classes={"zhaquirks.tuya.ts0601_garage.TuyaGarageSwitchTO"})
```

Comment:
Does this actually match ?
Have you checked that the component appears in the UI ?

From the code, you should be using "ts0601_garage.TuyaGarageSwitchTO" without the begining.

Reply:
> without the begining.

That's only when using a custom quirk, right? When the quirk gets merged into `zha-quirks`, we need to use the `zhaquirks.` prefix here.

Reply:
I doubt it. quirk_class is defined here : https://github.com/home-assistant/core/blob/4511e91cfd58fd097e1680d67c48bf22c25a47ad/homeassistant/components/zha/core/device.py#L125

As far I understand, the module will always be a single word without dot in it. You will not get the full path.

Reply:
Mhh wait indeed I have the full path when not using a custom quirk. I guess I never actually looked at working devices. I guess we'll have to find a better way to fix this then.

Reply:
```suggestion
@MULTI_MATCH(cluster_handler_names="tuya_manufacturer_garage", quirk_classes={"zhaquirks.tuya.ts0601_garage.TuyaGarageSwitchTO"})
```

I've been attempting this and found that the code in the PR matches but causes an IndexError: list index out of range. After a bunch of debugging I believe this is because it is looking for the "contact_sensor" attribute on the "Basic" cluster rather than the TuyaGarageManufCluster cluster. I might be off base here, but the suggestion provided gets me a working sensor in the UI. (Testing with a clean 2023.06.01 + components/zha copied over into custom_components)

For code cleanliness, is there a way to pull these strings from the quirk definition rather than hardcoding?